opencensus/exporters/trace/zipkin/internal/zipkin_exporter.cc

```diff
@@ -0,0 +1,386 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "opencensus/exporters/trace/zipkin/zipkin_exporter.h"
+
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <iostream>
+
+#include <curl/curl.h>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "opencensus/trace/exporter/attribute_value.h"
+
+namespace opencensus {
+namespace exporters {
+namespace trace {
+
+namespace {
+
+constexpr char kZipkinLib[] = "zipkin/2.0";
+constexpr char ipv4_loopback[] = "127.0.0.1";
+constexpr char ipv6_loopback[] = "::1";
+
+std::string SerializeMessageEvent(
+    const ::opencensus::trace::exporter::MessageEvent& event) {
+  return absl::StrCat(
+      event.type() == ::opencensus::trace::exporter::MessageEvent::Type::SENT
+          ? "SENT"
+          : "RECEIVED",
+      "/", std::to_string(event.id()), "/",
```

Comment:
What does event.id() return that can't be StrCatted without the explicit to_string()?

Reply:
Fixed.